idl/github.com/uber/cadence/shared.thrift

```diff
@@ -1462,3 +1462,91 @@ struct HistoryBranch{
   20: optional string branchID
   30: optional list<HistoryBranchRange>  ancestors
 }
+
+enum BatchOperationType{
+    Terminate
+    RequestCancel
+    Signal
+}
+
+struct BatchSignalRequest{
+    10: optional string signalName
+    20: optional binary signalData
+}
+
+struct BatchTerminateRequest{
+    10: optional bool applyOnChildren
+}
+
+struct BatchCancelRequest{
+    10: optional bool applyOnChildren
+}
+
+struct StartBatchJobRequest{
+    10: optional string domain
+    20: optional SearchAttributes searchAttributes
+    30: optional BatchOperationType operationType
+    40: optional string sourceQuery
+    50: optional string reason
+    60: optional BatchSignalRequest signalRequest
+    70: optional BatchTerminateRequest terminateRequest
+    80: optional BatchCancelRequest cancelRequest
+    90: optional i64 rps
+    100: optional Header header
+    110: optional i32 startToCloseTimeoutSeconds
+    120: optional string requestID
+    130: optional binary identity
+
+}
+
+struct StartBatchJobResponse{
+    10: optional string jobID
+}
+
+struct StopBatchJobRequest{
+    10: optional string domain
+    20: optional string jobID
+}
+
+struct DescribeBatchJobRequest{
+    10: optional string domain,
+    20: optional string jobID
+}
+
+struct DescribeBatchJobResponse{
+    10: optional i64 totalEstimate
+    20: optional i64 successCount
+    30: optional i64 failureCount
+    40: optional WorkflowExecutionCloseStatus status
```

Comment:
What is this? Why batch request has workflowexecutionclosestatus?

Reply:
I reused this enum. For a better name we can create the same one like BatchJobCloseStatus. But essentially they are the same because batch job is essentially a workflow, and we don't need to map from BatchJobCloseStatus to WorkflowExecutionCloseStatus if we reusing it.

Reply:
I think a separate enum is preferable for clarity.